src/os_macosx.m

```diff
@@ -482,12 +475,12 @@ - (void) sound:(NSSound *)sound didFinishPlaying:(BOOL)flag
 {
     @autoreleasepool
     {
-	NSSound *sound = sounds_list[[NSNumber numberWithLong:sound_id]];
+    NSNumber *_sound_id = [NSNumber numberWithLong:sound_id];
+	NSSound *sound = sounds_list[_sound_id];
 	if (sound != nil)
 	{
-	    // Stop the sound. No need to release it because the delegate will do
-	    // it for us.
 	    [sound stop];
+        [sounds_list removeObjectForKey:_sound_id];
```

Comment:
I think this should be done at the delegate per the other [comment](https://github.com/vim/vim/pull/12337/files#r1835587863) mentioned.